common/src/express/types.ts

```diff
@@ -0,0 +1,39 @@
+import { Schema, Get } from '../api-types'
+import { PathParams } from 'express-serve-static-core'
+import express from 'express'
+
+type TypedHandlerForMethod<Method extends Schema.Method> = express.RequestHandler<
+  Get.Request.Parameters<Method>,
+  Get.Response.Body<Method>,
+  Get.Request.Body<Method>,
+  Get.Request.Query<Method>
+>
+
+export type TypedHandler<
+  Endpoint extends Schema.Endpoint,
+  Path extends Get.Path<Endpoint>,
+  MethodName extends Schema.MethodName = Schema.MethodName
+> = TypedHandlerForMethod<Get.ChildMethod<Endpoint, Path, MethodName>>
+
+interface TypedMatcher<
+  Endpoint extends Schema.Endpoint,
+  ReturnType,
+  MethodName extends Schema.MethodName = Schema.MethodName
+> {
+  <Path extends Get.Path<Endpoint> & PathParams>(
+    path: Path,
+    ...handlers: Array<TypedHandler<Endpoint, Path, MethodName>>
+  ): ReturnType
+}
+
+export interface TypedRouter<Endpoint extends Schema.Endpoint> {
+  all: TypedMatcher<Endpoint, this, 'all'>
```

Comment:
`Matcher` in the name `TypedMatcher` doesn't make any sense re: web development and API servers. The names of types that include `API`, `Endpoint`, `Router` etc make sense. `TypedMatcher` is too abstract, can we come up with a better name for this type?